lib/codecs/XsdDateTimeEncoder.js

```diff
@@ -43,7 +49,13 @@ export class XsdDateTimeEncoder extends CborldEncoder {
     return [new Token(Type.array, entries.length), entries];
   }
 
-  static createEncoder({value} = {}) {
+  static createEncoder({value, transformer} = {}) {
+    if(transformer.stringTable.has(value)) {
+      return new XsdDateTimeEncoder({
+        value,
+        compressionMap: transformer.stringTable
+      });
+    }
```

Comment:
The current implementation here hits the string table map 3 times here. Suggestion:

```suggestion
    const id = transformer.stringTable.get(value);
    if(id !== undefined) {
      return new XsdDateTimeEncoder({value, id});
    }
```